### ISSUE TITLE
Error: unable to extract n-function name

### ISSUE THREAD

**Issue by thenhawke** [kkdai/youtube#235]
Library has started throwing an "unable to extract n-function name" error when getting the stream URL on some videos. It looks like the regex for finding the function needs to be updated but I can't seem to get it right. 

There's a recent commit to youtube-dl that seems to fix this exact issue if that's helpful:
https://github.com/ytdl-org/youtube-dl/commit/1e677567cd083d43f55daef0cc74e5fa24575ae3

Code that throws error:
```
package main

import (
	"fmt"
	"log"

	"github.com/kkdai/youtube/v2"
)

func main() {
	client := youtube.Client{}

	video, err := client.GetVideo("djV11Xbc914")
	if err != nil {
		log.Fatal(err)
	}

	formats := video.Formats.WithAudioChannels()

	url, err := client.GetStreamURL(video, &formats[0])
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(url)
}
```

**Comment by paij0se** [comment]
fr

**Comment by wellmoon** [comment]
base.js has changed, 
line 2000: `b=a.get("n"))&&(b=**Apa[0]**(b)`
the nFunctionNameRegexp need update, 
find the function name from the first param of array Apa at line 7242 `var Apa=[hha];`

**Comment by hamzah-hayat** [comment]
Looks like this was fixed by https://github.com/kkdai/youtube/pull/236
I run a music bot which was throwing the error over the past few days but updating to github.com/kkdai/youtube/v2 v2.7.8 fixed them.

**Comment by thenhawke** [comment]
Fixed by #236
